### PR TITLE
Upstream, a test's tags are an array of strings now.

### PR DESCRIPTION
**Pull request by doertedev**
Signed-off-by: Stefan Staudenmeyer <stefan.staudenmeyer@instana.com>

### CODE REVIEW

**Comment by croman** [comment]
Can you add a documentation link for this change? @doertedev

**Comment by doertedev** [comment]
https://www.statuscake.com/api/Tests/Get%20All%20Tests.md << see response example beneath.

**Comment by croman** [comment]
From what I see, inserting / updating a test accepts a string field **TestTags**, but "Get all tests" returns a string array field called **Tags**. You are changing the **TestTags** type but I think you should add a **Tags** field instead.

**Comment by doertedev** [comment]
I still don't really get some of your tests, but everything's green on my end now. Can you confirm?

File: fixtures/tests_all_ok.json [hunk 1]
```diff
@@ -5,6 +5,8 @@
       "TestType": "HTTP",
       "WebsiteName": "www 1",
       "ContactGroup": null,
+      "TestTags": "tag1,tag2",
```
Comment (croman) [inline fixtures/tests_all_ok.json:8]: This is not needed, the "Get All Tests" doesn't return the "TestTags" field.

File: fixtures/tests_all_ok.json [hunk 2]
```diff
@@ -15,6 +17,8 @@
       "TestType": "HTTP",
       "WebsiteName": "www 2",
       "ContactGroup": null,
+      "TestTags": "tag3,tag4",
```
Comment (croman) [inline fixtures/tests_all_ok.json:20]: This is not needed, the "Get All Tests" doesn't return the "TestTags" field.

File: tests_test.go [hunk 3]
```diff
@@ -157,6 +159,8 @@ func TestTests_All(t *testing.T) {
 		ContactID:   1,
 		Status:      "Up",
 		Uptime:      100,
+		TestTags:    "tag1,tag2",
```
Comment (croman) [inline tests_test.go:162]: This is not needed, the "Get All Tests" doesn't return the "TestTags" field.

File: tests_test.go [hunk 4]
```diff
@@ -168,6 +172,8 @@ func TestTests_All(t *testing.T) {
 		ContactID:   2,
 		Status:      "Down",
 		Uptime:      0,
+		TestTags:    "tag3,tag4",
```
Comment (croman) [inline tests_test.go:175]: This is not needed, the "Get All Tests" doesn't return the "TestTags" field.

**Review by croman — CHANGES_REQUESTED** [review]
Looks good, I think you need to delete "TestTags" usage when using the "Get All Tests" API, but then it's good to merge.

**Pull request closed without merging**